### ISSUE TITLE
About reimplementing result of experiments

### ISSUE THREAD

**Issue by hhhzzj** [jeromerony/fast_adversarial#1]
Hi jeromerony, recently i read your paper, and i am so interested in your work. I reimplement the experiments with this github and meet some problem.

1. In mnist_noteboook_example.ipynb, `model_url = 'https://www.dropbox.com/s/9onr3jfsuc3b4dh/mnist.pth?dl=1'`  the trained model is identical to adversarially trained model what you give in the bottom of README.
2. In your paper(sec 4.), base classifier obtain 85.51% on CIFAR-10, and the model's architecture and hyperparameters is identical to C&W attack's model, but the later one obtain 80% in arXiv:1608.04644.
3. I run mnist_noteboook_example.ipynb and replace dataset to test set. However, i get this result, which is lower than the experiment in the paper. How should i do to obtain the experiment result.
![9c27f53f20002a4b2726a3b7ec24842](https://user-images.githubusercontent.com/37574973/56861118-937bf800-69d0-11e9-876f-15710674963f.png)
Thank you so much!!!

**Comment by jeromerony** [comment]
Hi, thanks for the interest in our work.
To answer your questions in order:
1. I am not sure what the problem is: it is indeed intended that the example is on the adversarially trained model.
2. The architecture is the same and the hyper-parameters, to the extent of what was mentioned in the paper you cite are the same, however we might have used more augmentation than they did. I am not sure as to why there was such a difference.
3. The experiments in the paper were done on the first 1000 examples of the test set so the difference might come from that. Also, the experiment should be run with cudnn.deterministic for better reproducibility otherwise you might get inconsistent results across runs. Which experiment result are you trying to reproduce exactly?

**Comment by hhhzzj** [comment]
Sorry for misunderstanding.
1. I mean that in mnist_noteboook_example.ipynb, the model should be a trained model rather than an adversarially trained model.
2. Can you upload your augmented models to reproduce better the expriments? Thank you so much.
3. I actually did this experiments on the first 1000 examples, and i am trying to reproduce Table 1. in your paper.

**Comment by hhhzzj** [comment]
Hi, i have a question that the computation of attack success rate, and i did not find a clear definition of it. Because i find that if `(pred_cw100 != y).float().mean().item() * 100`  instead of  `(pred_cw100 != pred_orig).float().mean().item() * 100`  i can obtain 100% ASR.

 Besides, if i replace `labels` with the prediction of clean image in `ce_loss = F.cross_entropy(logits, labels, reduction='sum')` i also obtain 100% ASR for DDN.

**Comment by jeromerony** [comment]
I think the rational behind putting pred_orig was that the model was initially making the correct prediction on the samples for MNIST. Of course this is not always the case, especially when testing with 1000+ samples and more complicated datasets.
The attack success rate has many definitions depending on the context, but in our case it is indeed about measuring the portion of samples misclassified by the model, even if they have a 0 perturbation (meaning they are originally misclassified). So it would indeed be `(pred_cw100 != y).float().mean().item() * 100`.
